### ISSUE TITLE
[Feature] disable video creation after context/page creation

### ISSUE THREAD

**Issue by LanderBeeuwsaert** [microsoft/playwright#4821]
Let us know what functionality you'd like to see in Playwright and what is your use case.
We use video creation in playwright in our CI environment to easily see what went wrong if a test fails.
so during context creation, we do:

    if (this.createVideo) {
      context.recordVideo = { dir: this.storageUploadPath };
    }
    this.context = await this.browser.newContext(context);

However, we are only interested in the video if the test actually went wrong.
If the test went well, the creation of the video on page/context close only takes (quite a lot) of time away that's not useful.
It would therefor be useful if we could change the setting just before page/context close to say that "nah, don't create a video anyway"

Do you think others might benefit from this as well?
I would think so.

**Comment by mxschmitt** [comment]
Which test runner are you using? Seems for me that this makes more sense to be implemented in the user land. It wouldn't bring any performance benefit during the test, since it needs to record it anyway and if a test fails, it stops. If you are using [playwright-test](https://github.com/microsoft/playwright-test) - playwright official test runner - you would be able to implement such a logic yourself or file an issue that it gets integrated into the project like the `screenshotOnFailure` functionality.

It's mostly about deleting the video files when the tests are passing if I understood it correctly. A `videoOnFailure` functionality in the [playwright-test](https://github.com/microsoft/playwright-test) project sounds like a great idea.

**Comment by LanderBeeuwsaert** [comment]
Hi, thanks for the reply,
Well in one project we're using playwright-jest but in the other we're using vanilla playwright (we want more control there to have a pool of contexts running in parallel).

What we've noticed is the following:
1. during the test, ffmpeg is creating the frames, and this increases the CPU load significantly; but doesn't slow down the test execution. As you say, indeed, this can't be eliminated anyway.

2. After the test, on context close (or page close, I'm not sure) the video file is built from the recorded frames.
This takes A LOT of time, +- the same amount of time as the test itself, and I suppose a lot of CPU and disk read/writes.
All this is CPU & Disk load that could be avoided by being able to cancel creation of the video just before we close the context/page.
=> deleting the file after creation would not work as a solution for this.

**Comment by mxschmitt** [comment]
Yup for 2 I agree there too. (Afaik only Chromium relies on ffmpeg in the end and the others do it somehow else). Could be added improve that behaviour.

Maybe as a spec proposal to be discussed [`await page.close({ saveVideos: false })`](https://playwright.dev/docs/api/class-page?_highlight=close#pagecloseoptions) which drops the videos in the end and does not process them.

**Comment by LanderBeeuwsaert** [comment]
Jep, that would be the best solution.

**Comment by pavelfeldman** [comment]
(2) seems like a problem in itself, it does not happen on page close, it should be happening in a streaming manner.

**Comment by tymfear** [comment]
+1 for canceling video creation option

**Comment by nmann4** [comment]
Another +1. I would also be interested in this feature. I have some long running tests and I've noticed that closing the context/page can take a significant amount of time when video recording is enabled.

**Comment by gkushang** [comment]
➕  **+1** on closing context without processing video-frames. We are seeing very slow executions, and I see it may improve the performance.

**Comment by niyarlatotep** [comment]
+1

**Comment by rwoll** [comment]
Folded https://github.com/microsoft/playwright/issues/4821 into here which requests being able to start, stop, and then start a video again on a page.

**Comment by hyphenized** [comment]
+1 for being able to record on demand, would be useful to skip recording altogether while running certain fixtures

**Comment by marcusNumminen** [comment]
+1 for apis that we can manually control the video recording

**Comment by mikestopcontinues** [comment]
I'm interested in this feature for more easily automating video generation. Not in a test context, but rather to ensure that documentation videos stay up-to-date with what the app looks like. Right now, it requires too much stitching.

**Comment by shrektheking69** [comment]
@mikestopcontinues Can I ask what stitching you had to do to handle this? 
We are facing the same situation and could not come up with a good solution.

**Comment by mikestopcontinues** [comment]
@shrektheking69 I'm actually posting in support of a customer I have at [BrowserCat](https://www.browsercat.com). Since you can't guarantee the exact moment you would clip the video, I offered them the following suggestion:

1. Record the entire session.
2. Throw a 100% black overlay over the screen when you _don't_ want to record.
3. Use `ffmpeg` to clip every black frame from the video. (It as a `blackdetect` filter built in.)
4. Voila!

Yes, this sucks, but it's the best I could figure out at the moment.

If you're willing to work with Puppeteer, it supports recording multiple videos. This is also good news for Playwright, because it means the browser already supports flexible video recording.

**Comment by Sego921** [comment]
Please complete this task, without it it is impossible to correctly implement the video recording functionality
